src/clients/advanced/TokenLifespan.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import { Control, Controller } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import {
+  FormGroup,
+  Select,
+  SelectOption,
+  SelectVariant,
+  Split,
+  SplitItem,
+} from "@patternfly/react-core";
+
+import {
+  TimeSelector,
+  Unit,
+} from "../../components/time-selector/TimeSelector";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+
+type TokenLifespanProps = {
+  id: string;
+  name: string;
+  defaultValue: string;
+  control: Control<Record<string, any>>;
```

Comment:
```suggestion
  control: Control<FieldValues>;
```

Should give you the same type, closer matching the interface from React Form Hook.